types/wlr_wl_shell.c

```diff
@@ -18,32 +20,128 @@ static void shell_surface_pong(struct wl_client *client, struct wl_resource
 	surface->ping_serial = 0;
 }
 
-static void shell_surface_move(struct wl_client *client, struct wl_resource
-		*resource, struct wl_resource *seat, uint32_t serial) {
-	wlr_log(L_DEBUG, "TODO: implement shell surface move");
+static void shell_surface_move(struct wl_client *client,
+		struct wl_resource *resource, struct wl_resource *seat_resource,
+		uint32_t serial) {
+	wlr_log(L_DEBUG, "got shell surface move");
+	struct wlr_wl_shell_surface *surface = wl_resource_get_user_data(resource);
+	struct wlr_seat_handle *seat_handle =
+		wl_resource_get_user_data(seat_resource);
+
+	struct wlr_wl_shell_surface_move_event *event =
+		calloc(1, sizeof(struct wlr_wl_shell_surface_move_event));
+	if (event == NULL) {
+		wl_client_post_no_memory(client);
+		return;
+	}
+
+	event->client = client;
+	event->surface = surface;
+	event->seat_handle = seat_handle;
+	event->serial = serial;
+
+	wl_signal_emit(&surface->events.request_move, event);
```

Comment:
Reminder that you should have the rootston side of things listen to this and attach the view to the cursor. This may be further down in your changes, I'll delete this if I find out.